### ISSUE TITLE
mpv (with youtube-dl installed) cannot stream videos from youtube anymore

### ISSUE THREAD

**Issue by gbstan** [mpv-player/mpv#5193]
### mpv version and platform
>mpv --version
mpv 0.27.0 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
 built on Thu Dec  7 10:37:50 GMT 2017
ffmpeg library versions:
   libavutil       55.78.100
   libavcodec      57.107.100
   libavformat     57.83.100
   libswscale      4.8.100
   libavfilter     6.107.100
   libswresample   2.9.100
ffmpeg version: 3.4

>brew info mpv
mpv: stable 0.27.0 (bottled), HEAD
Media player based on MPlayer and mplayer2
https://mpv.io
/usr/local/Cellar/mpv/0.27.0_2 (57 files, 33.7MB) *
  Built from source on 2017-12-07 at 10:38:23 with: --with-bundle

Platform: **MacOS X High Sierra 10.13.1 (17B1003)**

### Reproduction steps
mpv https://www.youtube.com/watch?v=z9BPMjL44Aw

### Expected behavior
Streaming of the video from youtube

### Actual behavior
mpv https://www.youtube.com/watch?v=z9BPMjL44Aw
Playing: https://www.youtube.com/watch?v=z9BPMjL44Aw

**[ffmpeg] tls: IO Error: -9806**
**Failed to recognize file format.**

### Log file

mpv -v https://www.youtube.com/watch?v=z9BPMjL44Aw
[cplayer] Command line options: '-v' 'https://www.youtube.com/watch?v=z9BPMjL44Aw'
[cplayer] mpv 0.27.0 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Thu Dec  7 10:37:50 GMT 2017
[cplayer] ffmpeg library versions:
[cplayer]    libavutil       55.78.100
[cplayer]    libavcodec      57.107.100
[cplayer]    libavformat     57.83.100
[cplayer]    libswscale      4.8.100
[cplayer]    libavfilter     6.107.100
[cplayer]    libswresample   2.9.100
[cplayer] ffmpeg version: 3.4
[cplayer] 
[cplayer] Configuration: waf configure --prefix=/usr/local/Cellar/mpv/0.27.0_2 --enable-zsh-comp --enable-libmpv-shared --enable-html-build --confdir=/usr/local/etc/mpv --datadir=/usr/local/Cellar/mpv/0.27.0_2/share/mpv --mandir=/usr/local/Cellar/mpv/0.27.0_2/share/man --docdir=/usr/local/Cellar/mpv/0.27.0_2/share/doc/mpv --zshdir=/usr/local/Cellar/mpv/0.27.0_2/share/zsh/site-functions
[cplayer] List of enabled features: apple-remote asm atomics avutil-content-light-level avutil-icc-profile avutil-imgcpy-uc avutil-spherical bsd-fstatfs build-date cocoa coreaudio cplayer cplugins cuda-hwaccel debug-build encoding fchmod gl gl-cocoa glob glob-posix gnuc iconv is_ffmpeg jpeg lcms2 libass libass-osd libav libavcodec libavdevice libdl libm libmpv-shared macos-touchbar nanosleep optimize osx-thread-name plain-gl posix posix-or-mingw posix-spawn pthreads shm stdatomic termios videotoolbox-gl videotoolbox-hwaccel videotoolbox-hwaccel-new zlib zsh-comp
[global] config path: '' -> '/Users/guy-bartstan/.config/mpv'
[global] config path: 'encoding-profiles.conf' -/-> '/Users/guy-bartstan/.config/mpv/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -/-> '/Users/guy-bartstan/.mpv/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -/-> '/usr/local/bin/encoding-profiles.conf'
[global] config path: 'encoding-profiles.conf' -> '/usr/local/etc/mpv/encoding-profiles.conf'
[cplayer] Reading config file /usr/local/etc/mpv/encoding-profiles.conf
[global] config path: 'mpv.conf' -> '/Users/guy-bartstan/.config/mpv/mpv.conf'
[global] config path: 'config' -/-> '/Users/guy-bartstan/.config/mpv/config'
[global] config path: 'mpv.conf' -/-> '/Users/guy-bartstan/.mpv/mpv.conf'
[global] config path: 'config' -/-> '/Users/guy-bartstan/.mpv/config'
[global] config path: 'mpv.conf' -/-> '/usr/local/bin/mpv.conf'
[global] config path: 'config' -/-> '/usr/local/bin/config'
[global] config path: 'mpv.conf' -/-> '/usr/local/etc/mpv/mpv.conf'
[global] config path: 'config' -/-> '/usr/local/etc/mpv/config'
[cplayer] Reading config file /Users/guy-bartstan/.config/mpv/mpv.conf
[cplayer] Setting option 'screenshot-format' = 'png' (flags = 4)
[cplayer] Setting option 'screenshot-template' = '~/Downloads/mpv-screenshot%n' (flags = 4)
[cplayer] Setting option 'hwdec' = 'videotoolbox' (flags = 4)
[cplayer] Setting option 'profile' = 'opengl-hq' (flags = 4)
[cplayer] Setting option 'scale' = 'spline36' (flags = 4)
[cplayer] Setting option 'cscale' = 'spline36' (flags = 4)
[cplayer] Setting option 'dscale' = 'mitchell' (flags = 4)
[cplayer] Setting option 'dither-depth' = 'auto' (flags = 4)
[cplayer] Setting option 'correct-downscaling' = 'yes' (flags = 4)
[cplayer] Setting option 'sigmoid-upscaling' = 'yes' (flags = 4)
[cplayer] Setting option 'deband' = 'yes' (flags = 4)
[cplayer] Setting option 'slang' = 'eng,en,fr' (flags = 4)
[cplayer] Setting option 'alang' = 'eng,en,fr' (flags = 4)
[cplayer] Setting option 'v' = '' (flags = 8)
[global] config path: 'input.conf' -/-> '/Users/guy-bartstan/.config/mpv/input.conf'
[global] config path: 'input.conf' -/-> '/Users/guy-bartstan/.mpv/input.conf'
[global] config path: 'input.conf' -/-> '/usr/local/bin/input.conf'
[global] config path: 'input.conf' -/-> '/usr/local/etc/mpv/input.conf'
[global] config path: 'scripts' -/-> '/Users/guy-bartstan/.config/mpv/scripts'
[global] config path: 'scripts' -/-> '/Users/guy-bartstan/.mpv/scripts'
[global] config path: 'scripts' -/-> '/usr/local/bin/scripts'
[global] config path: 'scripts' -/-> '/usr/local/etc/mpv/scripts'
[global] config path: 'watch_later' -> '/Users/guy-bartstan/.config/mpv/watch_later'
[cplayer] Playing: https://www.youtube.com/watch?v=z9BPMjL44Aw
[ffmpeg] Opening https://www.youtube.com/watch?v=z9BPMjL44Aw
[ffmpeg] Mime-type: 'text/html; charset=utf-8'
[ffmpeg] Stream opened successfully.
[cache] Cache size set to 150000 KiB (75000 KiB backbuffer)
[demux] Trying demuxers for level=normal.
[ffmpeg] tls: IO Error: -9806
[cache] EOF reached.
[lavf] No format found, try lowering probescore or forcing the format.
[demux] Trying demuxers for level=unsafe.
[lavf] No format found, try lowering probescore or forcing the format.
[cache] Terminating cache...
[cache] Cache exiting...
[cplayer] Opening failed or was aborted: https://www.youtube.com/watch?v=z9BPMjL44Aw
[cplayer] finished playback, unrecognized file format (reason 4)
[cplayer] Failed to recognize file format.
[cplayer] 
[cplayer] 
[cplayer] Exiting... (Errors when loading file)
[osx] Exiting...

**Comment by sfan5** [comment]
You need to compile with `lua` support enabled.

**Comment by lachs0r** [comment]
Yup.

**Comment by gbstan** [comment]
Thanks. Can you give more details on how to do this using brew please?

**Comment by gbstan** [comment]
Ok. Found it on https://gist.github.com/doole/f84c90126eb6bdacf403

**Comment by AirPort** [comment]
Homebrew switched the default for lua to 5.3, which is incompatible with mpv's scripts. Manually install lua@5.1 or lua@5.2 beforehand with brew or just wait for this PR to be merged: https://github.com/Homebrew/homebrew-core/pull/21389

**Comment by gbstan** [comment]
Thanks AirPort. I have tried to compile with lua support but that didn't work:
brew install --with-lua --with-bundle mpv
Warning: mpv: this formula has no --with-lua option so it will be ignored!

Do you know if mpv will support lua 5.3 or do I really have to manually go to lua@5.2?

**Comment by sfan5** [comment]
5.3 [will not be supported](https://github.com/mpv-player/mpv/issues/4309#issuecomment-292928797), so you'll need to install either 5.2 or 5.1

**Comment by gbstan** [comment]
Ok. I see on brew that lua@5.1 and luajit are both available. Which one is better to install? lua@5.1 or luajit? I didn't need to do this before, so any help here would be useful for me.

Once I have installed lua with brew, e.g. with brew install luajit

Should I just install mpv this way then?: brew install --with-lua --with-bundle mpv

**Comment by AirPort** [comment]
`--with-lua` isn't needed and it doesn't even exists anymore, `brew install --with-bundle` should be enough. If it still doesn't work try with `brew install --build-from-source --with-bundle` (it's been a while since I last used brew for mpv, so I don't remember if the `--with-bundle` is enough to trigger an install from source).

**Comment by vitorgalvao** [comment]
@gbstan In your situation I’d simply wait a few hours. If you’re having this much trouble, it’s likely that by the time you fix it it has been fixed on HBs side as well and your time has been mostly wasted.

The creator of the PR mentioned by @AirPort is a former HB maintainer and it’s already decided that they will merge it; they’re just fixing another bug first.

> so I don't remember if the `--with-bundle` is enough to trigger an install from source).

I think it is.

**Comment by wm4** [comment]
There's a short FAQ entry about the Lua thing: https://github.com/mpv-player/mpv/wiki/FAQ#Why_does_mpv_not_support_Lua_53

**Comment by gbstan** [comment]
Thank you all for your help. I will wait until this is merged then and mpv dependencies are fixed.

**Comment by ilovezfs** [comment]
See my comments here: https://github.com/mpv-player/mpv/issues/5080#issuecomment-350411768

TL;DR I've merged the 5.1 PR :peace_symbol: :dove: :v:
